tests/unit/plugins/modules/net_tools/test_dnsimple_info.py

```diff
@@ -0,0 +1,25 @@
+__metaclass__ = type
+import pytest
+import traceback
+import sys
+
+try:
+    import dnsimple_info
+    from requests import Request, Session
+    from community.general.plugins.modules.net_tools import dnsimple_info
+except ImportError:
+    HAS_ANOTHER_LIBRARY = False
+    ANOTHER_LIBRARY_IMPORT_ERROR = traceback.format_exc()
+else:
+    HAS_ANOTHER_LIBRARY = True
+
+# First and required function is building our URL and getting a response,
+# so we must test that we can get a response and that it shows a failed
+# auth since our test case has no valid key/ID. Alternatively it may show:
+# Too many requests
```

Comment:
You probably want to use httpmock for this. Talking to real servers is a bad idea for unit tests.

See https://github.com/ansible-collections/community.general/blob/main/tests/unit/plugins/modules/source_control/gitlab/test_gitlab_deploy_key.py and https://github.com/ansible-collections/community.general/blob/main/tests/unit/plugins/modules/source_control/gitlab/gitlab.py for how this could be done.